CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+# Contribution Guidelines
+
+Busy-Beaver welcomes any, and all, contributions. Every little bit helps!
+
+## Architecture Overview
+
+Busy-Beaver is a Python application with a Slack frontend. The application consists of a set of REST endpoints with integrations to various services (GitHub, Slack, YouTube, Trello, etc) using public APIs / 3rd party libraries.
+
+While the application is currently a monolith, it is built with Service-Oriented Architecture (SOA) in mind. API endpoints are implemented using Flask blueprints and services are integrated using the Adapter / Facade pattern.
+
+Busy-Beaver tasks are kicked off through `curl` requests scheduled via CRON.
+
+### GitHub Activity Workflow
+
+The following diagram shows a high-level workflow of the GitHub activity features:
+<img src="assets/architecture.png" width=800 />
+
+## Development Environment
+
+It is recommended that users create a personal Slack workspace to use for bot development. This will allow for independent development without having to wait for project maintainers to grant access to the Busy-Beaver development Slack.
+
+### Slack Integration
+
+1. [Create a Slack workspace](https://get.slack.help/hc/en-us/articles/206845317-Create-a-Slack-workspace)
+1. [Create a Slack App](https://api.slack.com/apps) and set the development workspace to the workspace from the previous step
+1. Configure Slack app settings (defined below)
+1. Install application in workspace
+1. Locate the `Bot User OAuth Access Token`, we will need that later on
+
+#### Slack Settings
+
+- Add a bot user
```

Comment:
Some screenshots to help guide contributors
![add-bot-user-1](https://user-images.githubusercontent.com/7044060/54065757-0f7c6f80-41da-11e9-9565-803872b489aa.png)
![add-bot-user-2](https://user-images.githubusercontent.com/7044060/54065758-0f7c6f80-41da-11e9-9b51-ab55e46469e4.png)

Reply:
That's awesome!